serde_derive/src/de.rs

```diff
@@ -2143,16 +2141,16 @@ fn deserialize_identifier(
     ) = if collect_other_fields {
         (
             Some(quote! {
-                let __value = _serde::__private::de::Content::String(_serde::__private::ToString::to_string(__value));
+                let __value = _serde::de::buffer::BufferVisitor::new().visit_string::<_serde::de::value::Error>(_serde::__private::ToString::to_string(__value)).unwrap();
             }),
             Some(quote! {
-                let __value = _serde::__private::de::Content::Str(__value);
+                let __value = _serde::de::buffer::BufferVisitor::new().visit_borrowed_str::<_serde::de::value::Error>(__value).unwrap();
             }),
             Some(quote! {
-                let __value = _serde::__private::de::Content::ByteBuf(__value.to_vec());
+                let __value = _serde::de::buffer::BufferVisitor::new().visit_byte_buf::<_serde::de::value::Error>(__value.to_vec()).unwrap();
             }),
             Some(quote! {
-                let __value = _serde::__private::de::Content::Bytes(__value);
+                let __value = _serde::de::buffer::BufferVisitor::new().visit_borrowed_bytes::<_serde::de::value::Error>(__value).unwrap();
             }),
```

Comment:
Using the `BufferVisitor` here is a kind of hack. This is where the `From` implementations would be useful.